client/state/selectors/utils/index.js

```diff
@@ -15,26 +13,25 @@ import getPublicizeConnection from 'state/selectors/get-publicize-connection';
  * passed as a parameter.
  * Thus, it returns richest array adding and crossing information for each action.
  *
- * @param {Object} state - Global state tree
- * @param {Array} postShareActions - share actions of a post
- * @return {Array} richest post actions array
+ * @param  {object} state - Global state tree
+ * @param  {Array} postShareActions - share actions of a post
+ * @returns {Array} richest post actions array
  */
 export function enrichPublicizeActionsWithConnections( state, postShareActions ) {
 	return map(
 		postShareActions,
 		( { ID, connection_id, message, result, share_date, status, external_url: url } ) => {
 			const connection = getPublicizeConnection( state, connection_id );
-			const actionDate = moment( share_date );
+			const actionDate = new Date( share_date );
 
 			return {
 				ID,
-				connectionId: get( connection, 'ID' ),
+				connectionId: connection?.ID,
 				connection,
-				connectionName: get( connection, 'external_display' ),
+				connectionName: connection?.external_display,
 				message,
 				result,
-				service: get( connection, 'service', 'twitter' ),
-				shareDate: actionDate.format( 'llll' ),
+				service: connection?.service?.twitter,
```

Comment:
The correct expression here is `connection?.service ?? 'twitter'`. I'm fixing it locally and will push a commit. Discovered this bug when searching where have all the social icons gone 🙂

Reply:
Ahhh, good catch! I missed that it was a third parameter and not a string array 🤦‍♂